playground/src/state-view/state-view.js

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 The AMPHTML Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const JSONTreeView = require('json-tree-view');
+require('json-tree-view/example/build/devtools.css');
+require('./state-view.scss');
+import events from '../events/events.js';
+import * as Button from '../button/button.js';
+import * as Preview from '../preview/preview.js';
+import FlyIn from '../fly-in/base.js';
+
+
+export function createStateView(target, trigger) {
+  return new StateView(target, trigger);
+}
+
+export const EVENT_AMP_BIND_REQUEST_STATE =
+  'event-amp-bind-request-state';
+
+class StateView extends FlyIn {
+  constructor(target, trigger, preview) {
+    super(target);
+
+    this.target = target;
+    this.trigger = Button.from(trigger, this.requestState.bind(this));
+
+    // Content container
+    this.container = document.createElement('div');
```

Comment:
When scanning through this I found `container` to be irritating. What about naming this `content` instead?

Reply:
While going further through it I saw that the base class already defines `content` - as stated later: it probably makes sense to make the APIs a little more useful here 🙂